File: packages/core-java/src/main/java/com/openpasskey/core/CoseConstants.java
package com.openpasskey.core;

/**
 * COSE algorithm and key type constants used in WebAuthn.
 */
public final class CoseConstants {
    private CoseConstants() {}

    // Key types
    public static final int KTY_EC2 = 2;
    public static final int KTY_MLDSA = 8;
    public static final int KTY_COMPOSITE = 9;

    // Algorithms
    public static final int ALG_ES256 = -7;
    public static final int ALG_MLDSA65 = -49;
    public static final int ALG_COMPOSITE_MLDSA65_ES256 = -52;

    // COSE key map labels
    public static final int LABEL_KTY = 1;
    public static final int LABEL_ALG = 3;
    public static final int LABEL_EC_X = -2;
    public static final int LABEL_EC_Y = -3;
    public static final int LABEL_PUB = -1;
}

File: packages/core-dotnet/CoseConstants.cs
namespace OpenPasskey.Core;

public static class CoseConstants
{
    // Algorithm identifiers
    public const int AlgES256 = -7;
    public const int AlgMLDSA65 = -49;
    public const int AlgCompositeMLDSA65ES256 = -52;

    // Key type identifiers
    public const int KtyEC2 = 2;
    public const int KtyMLDSA = 8;
    public const int KtyComposite = 9;

    // ML-DSA-65 public key size (FIPS 204)
    public const int MLDSA65PubKeySize = 1952;

    // Uncompressed EC P-256 point: 0x04 || x(32) || y(32)
    public const int ECDSAUncompressedSize = 65;
}
